### PR TITLE
✨ Source Confluence: Added support for Confluence Server/Data Center

### PR DESCRIPTION
**Pull request by nobowned**
Adds Confluence Server support to the confluence-source actor/connector. Currently this connector only supports Confluence Cloud.

Please note that I was not able to add integration tests for this due to the inability to offer credentials to the private Confluence server I'm adding support for.

Fixes [36384 ](https://github.com/airbytehq/airbyte/discussions/36384) and [21657](https://github.com/airbytehq/airbyte/issues/21657)

### CODE REVIEW

File: airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml [hunk 1]
```diff
@@ -152,8 +162,19 @@ spec:
         description: 'Please follow the Jira confluence for generating an API token: <a href="https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/">generating an API token</a>.'
         airbyte_secret: true
         order: 1
-      domain_name:
-        title: "Domain name"
-        description: "Your Confluence domain name"
+      base_url:
+        title: "Base URL"
+        description: "Your Confluence base URL"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml:167]: In theory you don't need to change this variable as you need to continue to specify the server endpoint.

File: airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml [hunk 2]
```diff
@@ -41,13 +41,21 @@ definitions:
 
   requester:
     type: HttpRequester
-    url_base: https://{{ config['domain_name'] }}/wiki/rest/api/
+    url_base: "{{ config['base_url'] + '/rest/api/' if config.confluence_server else config['base_url'] + '/wiki/rest/api/' }}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml:44]:
```suggestion
    url_base: "{{ "https://" + config['domain_name'] + '/rest/api/' if config.confluence_server else "https:// + " + config['domain_name'] + '/wiki/rest/api/' }}"
```

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml:44]: This way it doesn't introduce a breaking change to existing connections and allow you to use the new option.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments but looks good!

File: airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml (hunk 1 above)
Comment (xactlance) [reply airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml]: Changed back.

File: airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml (hunk 2 above)
Comment (xactlance) [reply airbyte-integrations/connectors/source-confluence/source_confluence/manifest.yaml]: Done

**Review by xactlance — APPROVED** [review]
Done.
Edit: Oops I reviewed my own changes? I'm confused.

**Pull request closed without merging**